src/datetime.rs

```diff
@@ -171,6 +171,16 @@ impl crate::DateTime {
         self.to_chrono()
             .to_rfc3339_opts(chrono::SecondsFormat::AutoSi, true)
     }
+
+    /// Convert the given RFC 3339 format string to a [`DateTime`].
+    pub fn from_rfc3339(s: impl AsRef<str>) -> Result<Self, crate::de::Error> {
```

Comment:
Can we name this `parse_rfc3339_str`? That way it'll match `ObjectId::parse_str` and `Uuid::parse_str`. It's also somewhat closer to the familiar `chrono` API.

Reply:
Yup, done in https://github.com/mongodb/bson-rust/pull/317/commits/9496a4f83481f037935d44bbcbeeedbb882e383a.